design/trigger-backups-api-events.md

```diff
@@ -0,0 +1,35 @@
+# Trigger backups based on Kubernetes events 
+
+Triggering backups with Velero based on Kubernetes api events (Terminating, deleting, CrashLoopBackOff, etc..).
+
+## Goals
+
+- Rather than performing backups manually or by scheduling them, backups are created based on events.
+- Recovering the components of the cluster when they were deleted accidently. 
```

Comment:
Can you clarify this some? How is this different from what Velero already does?

Reply:
With Velero, we create backups manually or by scheduling them, what is different for waht Velero already does is that we trigger backups for the components that are being called to be deleted

Reply:
So you're envisioning that this would intercept a delete request and back up the components _before_ allowing them to delete?